transform/snowflake-dbt/models/mattermost/downloads.sql

```diff
@@ -4,39 +4,121 @@
   })
 }}
 
-WITH downloads AS (
+WITH outliers         AS (
     SELECT
-        logdate::date as logdate,
-        cip,
-        CASE
-            WHEN uri LIKE '%desktop%' THEN 'Desktop'
-            WHEN uri LIKE '%team%' and uri NOT LIKE '%appcenter%' THEN 'TE'
-            WHEN uri NOT LIKE '%team%' and uri NOT LIKE '%appcenter%' THEN 'EE'
-            else '?'
-        END AS type,
-    	COUNT(*) AS count,
-    	MAX(CASE
-    	    WHEN uri LIKE '%desktop%' THEN split_part(uri, '/', 3)
-    	    ELSE split_part(uri, '/', 2)
-    	END) AS version,
-    	MAX(useragent) AS useragent
-    FROM {{ source('releases', 'log_entries') }} 
-    WHERE ((uri LIKE '%desktop%')
-            OR (
-                uri LIKE '/6.%'
-                OR uri LIKE '/5.%'
-                OR uri LIKE '/4.%'
-                OR uri LIKE '/3.%'
-                OR uri LIKE '/2.%'
-                OR uri LIKE '/1.%')
-            )
-    {% if is_incremental() %}
+        logdate::DATE                                           AS logdate
+      , logtime
+      , cip
+      , uri
+      , bytessent
+      , abs(bytessent - avg(bytessent) OVER (PARTITION BY uri)) AS diff
+      , avg(bytessent) OVER (PARTITION BY uri)                  AS avg
+      , max(bytessent) OVER (PARTITION BY uri)                  AS max
+      , stddev(bytessent) OVER (PARTITION BY uri) * 1           AS std
+      , count(bytessent) OVER (PARTITION BY uri)                AS count
+      , CASE
+            WHEN abs(bytessent - avg(bytessent) OVER (PARTITION BY uri)) >
+                 stddev(bytessent) OVER (PARTITION BY uri) * 1 THEN TRUE
+            ELSE FALSE END                                      AS one_std_from_mean
+      , CASE
+            WHEN abs(bytessent - avg(bytessent) OVER (PARTITION BY uri)) >
+                 stddev(bytessent) OVER (PARTITION BY uri) * 2 THEN TRUE
+            ELSE FALSE END                                      AS two_std_from_mean
+    FROM {{ source('releases', 'log_entries') }} log_entries
+    WHERE (regexp_like(uri, '^\/[1-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}\/.*') OR
+           regexp_like(log_entries.uri, '^\/desktop\/[1-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}\/.*'))
+      AND bytessent > 0
+      AND status LIKE '2%'
+),
 
-        -- this filter will only be applied on an incremental run
-        AND logdate > (SELECT MAX(logdate) FROM {{ this }})
+     server_downloads AS (
+         SELECT
+             log_entries.logdate::DATE                                                             AS logdate
+           , log_entries.logtime
+           , log_entries.cip                                                                       AS ip_address
+           , log_entries.uri
+           , CASE
+                 WHEN regexp_like(log_entries.uri, '^\/[1-9]{1,3}[.][0-9]{1,3}[.][0-9]{1,3}\/.*') THEN 'server'
```

Comment:
Download Type vs Type is a little confusing from a naming convention standpoint. Curious does enterprise & team type always equal server and desktop is always app?

Reply:
@rbradleyhaas yes, servers are either team or enterprise, apps are always desktop, since mobile apps don't exist in the rlease.log_entries table. I agree re: confusing naming convention. Maybe download_type to category, so we have a category (app or server) and type (team, enterprise, desktop)?

Reply:
Love this idea. Completely makes sense too.